### ISSUE TITLE
Expand documentation on migrations

### ISSUE THREAD

**Issue by shounakmulay** [simolus3/moor#208]
I am running into a weird issue where database tables are created properly on the emulator. But when I run the app on a device, couple of tables do not get created. If I clear the app data from setting on the device and run the app again, everything works perfectly again. 
[Here is the SO question for more details.](https://stackoverflow.com/questions/58535749/some-sqlite-moor-tables-created-only-after-clearing-app-data-in-flutter)

**Comment by shounakmulay** [comment]
So the issue was not with moor or my code. As the app is in not released yet and I am still making changes to the database, I did not want make changes to schema version and provide a migration strategy. My understanding was that the database will get deleted on app uninstall and whatever changes made to the database will be the new schema for the app next time I install it. That is not the case. I am new to development and this might be a very obvious thing to others. I just changed the database file for now to solve this issue. 
I am closing this issue, but it would be nice to know what exactly happens to the database on uninstall and also what I did wrong.
Thanks!
